ipn/ipnlocal/local_test.go

```diff
@@ -1330,3 +1331,317 @@ func (rc *routeCollector) AdvertiseRoute(pfx netip.Prefix) error {
 	rc.routes = append(rc.routes, pfx)
 	return nil
 }
+
+func setSyspolicyHandlerForTest(tb testing.TB, h syspolicy.Handler) {
+	tb.Helper()
+	oldHandler := syspolicy.PolicyHandler
+	syspolicy.PolicyHandler = h
+	tb.Cleanup(func() {
+		syspolicy.PolicyHandler = oldHandler
+	})
+}
+
+type mockSyspolicyHandler struct {
+	t          *testing.T
+	key        syspolicy.Key
+	s          string
+	exitNodeID string
+	exitNodeIP string
+	bothKeys   bool
+	err        error
+}
+
+func (h *mockSyspolicyHandler) ReadString(key string) (string, error) {
+	if key != string(h.key) {
+		return "", h.err
+	}
+	if h.bothKeys { // when both keys are used
```

Comment:
This will never be reached when it matters, the key already won't match so it'll have returned `"", h.err` above.

It's probably easier to just have a pair of booleans that control if the ID can be looked up or the IP can be looked up. Either that or always return the keys as existing, and they just return `""` if unset for the test.